### PR TITLE
kojiapi: make repo gpgkey optional

### LINKED ISSUE #985: koji api: gpgkey is not required even though the openapi spec says it is
The subject says it all.

### PR DESCRIPTION
**Pull request by ondrejbudai**
The kojiapi shouldn't require it as we need to build images from
unsigned packages.

Fixes #985

### CODE REVIEW

**Review by teg — APPROVED** [review]
Thanks!

**Comment by teg** [comment]
This is looking good, but needs a rebase.

File: internal/kojiapi/server.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (server *Server) PostCompose(w http.ResponseWriter, r *http.Request) {
 		repositories := make([]rpmmd.RepoConfig, len(ir.Repositories))
 		for j, repo := range ir.Repositories {
 			repositories[j].BaseURL = repo.Baseurl
-			repositories[j].GPGKey = repo.Gpgkey
+			repositories[j].GPGKey = *repo.Gpgkey
```
Comment (Gundersanne) [inline internal/kojiapi/server.go:109]: Hm this might be nil now, no? Not sure what happens in go when you deref nil.

Comment (ondrejbudai) [reply internal/kojiapi/server.go]: You are right, thanks! Fixed.

**Comment by ondrejbudai** [comment]
Rebased and modified a test - now we build images both with a gpg key and without one.

**Review by teg — APPROVED** [review]
Nice! But requires rebase.

**Comment by ondrejbudai** [comment]
Rebased

**Review by larskarlitski — APPROVED** [review]
Nice! Thanks :)

**Comment by ondrejbudai** [comment]
ostree failure is fixed in #1027. F32 tests failed due to flakey mirrors.

**Pull request merged**